datameta/api/metadata.py

```diff
@@ -87,6 +88,8 @@ def put(request:Request):
     target_metadatum = resource_by_id(db, MetaDatum, metadata_id)
 
     body = request.openapi_validated.body
+    if body["isFile"] and not body["name"]:
```

Comment:
I think the name and any other mandatory string can never be empty, independent of whether it is a file or not. Is this checked somewhere @lkuchenb

Reply:
@cschu there actually is a difference between metadata and metadatasets, which is a bit confusing :laughing:

Reply:
> I think the name and any other mandatory string can never be empty, independent of whether it is a file or not. Is this checked somewhere @lkuchenb
They can. I believe in our last discussion about this we came to the conclusion that empty strings should in principle be valid values for mandatory metadata. Thus I opened #149 to at least cover this for file names, where it makes no sense to allow empty strings at all

Reply:
@KerstenBreuer It is however now caught in the spreadsheet based input. The `convert` endpoint that turns spreadsheets into JSON metadatasets now converts empty strings to `null` since #261

Reply:
I wasn't sure whether this needed to be changed as well as @lkuchenb only mentioned the metadatasets endpoint. That said, I felt that having the empty file name check here as well wouldn't hurt, but maybe I'm wrong...

Reply:
I just realized now that we're in the `PUT:/metadata` endpoint here and not in `PUT:/metadatasets` (which does not exist) 🙈 

This is the admin endpoint for changing the metadata definitions. It could also use some checking, but that's not the issue at hand. Please revert this insert.